### ISSUE TITLE
Need Service type indication 

### ISSUE THREAD

**Issue by segevere** [IETF-OPSAWG-WG/l3nm#66]
under vpn-service add attribute for defining the service type for example 

- end user service
- b2b
- mobile backhaul  - X1
- mobile backhaul - S2

etc.
so the Orchestration management system can apply different logic in service creation.

**Comment by boucadair** [comment]
I like this. 

We may use more specific description than "end user service" (e.g., data service, VoIP, etc.) 

If we can provide examples how this information can be used to infer the service creation logic, this would be great.

**Comment by segevere** [comment]
the service type indication can help the orchestration/engine that computes the service to apply unique behavior to service, for example. 
in mobile backhaul, for X1 service the engine the creates the service will use LDP or basic SR tunnels however for S2 services the engine will allocate  SR-TE tunnels with special pcep constraints which are not part of the service payload. In addition  we respect to type non exposed parameters of the service can be computed by the orchestration automatically, alarm thresholds, using BGP-PIC, using internal OAM etc.

**Comment by boucadair** [comment]
Wouldn't then make sense to define profiles for "activation-guidelines".  Service-type will need to be set to the profile id?

The definition of the profiles is obviously local to each provider.

**Comment by segevere** [comment]
Do yo mean adding to vpn-profiles? 
if could be 
    |     +--rw service-profile-identifier* [id]
    |     |  +--rw id    string 
so it can implicitly define the activation guidelines .
where do vpn-profiles defined? is it internal per provider and the standard is agnostic to it? however the provider can demand value per activation of service?

**Comment by boucadair** [comment]
> Do yo mean adding to vpn-profiles?

Yes.

> if could be
> | +--rw service-profile-identifier* [id]
> | | +--rw id string
> so it can implicitly define the activation guidelines .

Yes. I prefer to use an explicit name: s/service-profile-identifier/service-activation-identifier. 

> where do vpn-profiles defined? is it internal per provider and the standard is agnostic to it?

This is internal to a provider. We cannot mandate the exact service activation logic, but we allow for the module to signal an opaque parameter that has a special meaning within the provider.

> however the provider can demand value per activation of service?

That's it.

**Comment by segevere** [comment]
last question (I hope...) is there a way to publish the optional profiles to the user from the provider side?
in other words, how does the provider tell the user here are the optional activation-profiles that you can use?

**Comment by boucadair** [comment]
This can be done using, e.g., L3SM.

More generally, L3NM should be able to capture the VPN services **and capabilities** of a network and expose them to above layers as well. Some of these capabilities will be used to drive service negotiation/ordering which outcome will be captured using L3SM while others will be used locally by the network orchestrator. 

This is similar to the concern we discussed for max-prefixes (#34): such information should be ideally captured using L3SM.

**Comment by oscargdd** [comment]
Current profiles,  are kept for very specific things such as QoS, encryption, bfd and routing which apply at the **device** level. The logic in the controller is NOT impacted by the value of those parameters. What I would avoid is having that a propietary service type drives a different logic in the controller. That complete spoils the use of a standard. That would mean that an implementation cannot work for several operators. 

However, I think the diffent in behaviour that you want to consider is related to the requirement that

>  "the engine the creates the service will use LDP or basic SR tunnels however for S2 services the engine will allocate SR-TE tunnels with special pcep constraints which are not part of the service payload"

This is related on how to implement the transport between the VRFs. This can be a "transport-profile", whose specification could potentially be a separate piece of work. Or we can define some standard well-known profiles that apply to most cases

Summing, if the profile is limited to the transport, and we can define some well-known transport-profiles, such as:
- LDP preferred
- SR preferred
- TE preferred

this would be OK. That is, because it is limited to a given topic not covered by the draft (how the transport is done). And complementary with the service mapping work you can cover the part of requesting tunnels with special constraints for the service. 

In short, a service-profile as a general thing, I am against. A well-known transport profile, yes,

**Comment by boucadair** [comment]
Oscar, I think that we are talking about the same thing. Nevertheless, deciding whether to prefer LDP, SR, TE, SFC, use a mix thereof, etc.  is local to provider/controller. 

The key point from an interoperability standpoint is to be able to bind that parameter with a service.  The values don't need to be standardized. That is really policy-based and local to orchestrator and device config manager.

The description of the parameter should be clear this is about network-specific matters.

**Comment by segevere** [comment]
There are many parameters which the orchestration need in order to complete the configuration which are not applied via the interface. the transport policy/profile is one of them. 
If there is motivation to apply device profiles transparently (QoS, encryption, bfd and routing) why  not letting orchestration profile as well? 
anyway my initial request for service -type , the idea for transport profile , will be very helpful in order to give room for OSS-Orchestration SLA.
in our implementation for L3VPN Service we have package all the defaults and behavior parameters to a single profile, the profile name implicit gives an idea about the service nature (Gold Service, MB S1), however with respect to the interface profile parameters the orchestration can know 
- Which technology to choose
- monitoring to apply
- which defaults to use 
- policy to apply
- etc.

The motivation is reducing the amount of data that the OSS need to manage, , because in many cases the higher level controllers doesnt really care about the low level.

**Comment by oscargdd** [comment]
Transport preferences container has been added with pull request #87 of Lucia. This goes in ietf-02 version.

**Comment by oscargdd** [comment]
Pull request #106 from Med changest the transport preference into an ordered list.

**Comment by boucadair** [comment]
Generic policies are out of scope of the draft. We do cover the minimum regarding transport matters. Not sure if we need to elaborate further.
